### PR TITLE
Finish conversion of `w` commands to rzshell

### PR DESCRIPTION
**Pull request by ret2libc**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [x] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [x] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

- convert wo, wu, wd, ww, wt, wz to rzshell

**Test plan**

...

**Closing issues**

### CODE REVIEW

**Comment by ret2libc** [comment]
- [x] TODO: check codecov to see which of the changed commands need at least one test for coverage.

File: librz/core/cio.c [hunk 1]
```diff
@@ -1104,3 +894,202 @@ RZ_API RzCmdStatus rz_core_io_pcache_print(RzCore *core, RzIODesc *desc, RzCmdSt
...
+			break;
+		case RZ_CORE_WRITE_OP_SHIFT_LEFT:
+			buf[i] <<= hex[j];
+			break;
+		case RZ_CORE_WRITE_OP_SHIFT_RIGHT:
+			buf[i] >>= hex[j];
+			break;
+		default:
+			rz_warn_if_reached();
+			break;
```
Comment (wargio) [inline librz/core/cio.c:1021]: shouldn't we return or something?

**Review by wargio — APPROVED** [review]

File: librz/core/cio.c (hunk 1 above)
Comment (ret2libc) [reply librz/core/cio.c:1021]: I think it's alright to not do anything. This is anyway a programming error that will be printed by `rz_warn_if_reached`. When this case is reached, the buf will remain the same, which I think is fine.

File: test/db/cmd/cmd_wox [hunk 2]
```diff
@@ -33,7 +33,7 @@ wox 0xabcdef@!10
 ps 14
 EOF
 EXPECT=<<EOF
-helloworld\x00\x00\x00\x00
+è\x83Ǣ\x98Ŀ\x83\xcf\x00\x00\x00\x00
```
Comment (XVilka) [inline test/db/cmd/cmd_wox:36]: Looks like an unexpected result

**Review by XVilka — APPROVED** [review]

File: test/db/cmd/cmd_wox (hunk 2 above)
Comment (ret2libc) [reply test/db/cmd/cmd_wox:36]: I think *not* having helloworld is definitely right (the test does write "helloworld" and then xor it with "0xabcd"), however there is a difference in how `stable`-rizin and `dev`-rizin prints the string...

```
$ rizin -qv
0.3.4
$ rizin =
[0x00000000]> wx c3a883c7a298c4bf83cf
[0x00000000]> ps 14
\xc3\xa8\x83\xc7\xa2\x98\xc4\xbf\x83\xcf\x00\x00\x00\x00
[0x00000000]> 
```

```
$ ./build-dev/binrz/rizin/rizin -qv
0.4.0-git
$ ./build-dev/binrz/rizin/rizin =
[0x00000000]> wx c3a883c7a298c4bf83cf
[0x00000000]> ps 14
è\x83Ǣ\x98Ŀ\x83\xcf\x00\x00\x00\x00
[0x00000000]> 
```

**Comment by ret2libc** [comment]
Assuming it's green, I think this is ready.

File: test/db/cmd/cmd_wox (hunk 2 above)
Comment (wargio) [reply test/db/cmd/cmd_wox:36]: that is due how ps was working before (i.e. not working)

Comment (ret2libc) [reply test/db/cmd/cmd_wox:36]: So I guess it is alright as it is within this new branch, right?

Comment (wargio) [reply test/db/cmd/cmd_wox:36]: yes, but i would print using px here, not ps

**Pull request merged**